### PR TITLE
Do not check if CRL or MFT is stale, only validate the validity of th…

### PR DESCRIPTION
**Pull request by lolepezy**
Just remove the code that is checking stale CRL/MFT until the MFT EE certificate is expired.

Please note: this check is completely removed, so there will not be any special error or warning even after EE certificate becomes invalid. In case we need to have a error message on the MFT/CRL in question and not just a validity error on EE certificate, I will change the PR.

### CODE REVIEW

**Comment by timbru** [comment]
Hi,

I believe we should WARN only if the MFT or CRL are passed their nextUpdateTime. We should just never reject - unless the MFT EE certificate is expired, but that is checked elsewhere - not in the code you touched on this.

Tim

**Comment by lolepezy** [comment]
I changed the code so that the warning is still raised, but stale objects are never rejected.

File: src/main/java/net/ripe/rpki/commons/crypto/cms/manifest/ManifestCms.java [hunk 1]
```diff
@@ -159,11 +159,7 @@ private void checkManifestValidityTimes(ValidationOptions options, ValidationRes
         DateTime nextUpdateTime = getNextUpdateTime();
 
         if (now.isAfter(nextUpdateTime)) {
-            if (nextUpdateTime.plusDays(options.getMaxStaleDays()).isAfter(now)) {
-                result.warnIfTrue(true, ValidationString.MANIFEST_PAST_NEXT_UPDATE_TIME);
-            } else {
-                result.rejectIfTrue(true, ValidationString.MANIFEST_PAST_NEXT_UPDATE_TIME);
-            }
+            result.warnIfTrue(true, ValidationString.MANIFEST_PAST_NEXT_UPDATE_TIME);
```
Comment (erikrozendaal) [inline src/main/java/net/ripe/rpki/commons/crypto/cms/manifest/ManifestCms.java]: Please change to `result.warnIfTrue(now.isAfter(nextUpdateTime), ...)`.

File: src/main/java/net/ripe/rpki/commons/crypto/crl/X509CrlValidator.java [hunk 2]
```diff
@@ -70,11 +70,7 @@ private void checkNextUpdate(X509Crl crl) {
         DateTime nextUpdateTime = crl.getNextUpdateTime();
 
         if (now.isAfter(nextUpdateTime)) {
-            if (nextUpdateTime.plusDays(options.getMaxStaleDays()).isAfter(now)) {
-                result.warnIfTrue(true, ValidationString.CRL_NEXT_UPDATE_BEFORE_NOW, nextUpdateTime.toString());
-            } else {
-                result.rejectIfTrue(true, ValidationString.CRL_NEXT_UPDATE_BEFORE_NOW, nextUpdateTime.toString());
-            }
+            result.warnIfTrue(true, ValidationString.CRL_NEXT_UPDATE_BEFORE_NOW, nextUpdateTime.toString());
```
Comment (erikrozendaal) [inline src/main/java/net/ripe/rpki/commons/crypto/crl/X509CrlValidator.java]: Please change to `result.warnIfTrue(now.isAfter(nextUpdateTime), ...)`.

File: src/main/java/net/ripe/rpki/commons/crypto/cms/manifest/ManifestCms.java (hunk 1 above)
Comment (lolepezy) [reply src/main/java/net/ripe/rpki/commons/crypto/cms/manifest/ManifestCms.java]: Oh yes. Done.

**Pull request merged**